tests/Node/ModuleTest.php

```diff
@@ -90,6 +90,7 @@ public function __construct(Environment \$env)
     {
         parent::__construct(\$env);
 
+        \$macros = \$this->macros;
```

Comment:
Before this PR, the compiled constructor looked like this:

```php
    public function __construct(\Twig\Environment $env)
    {
        parent::__construct($env);
        $this->source = $this->getSourceContext();
        $this->parent = \false;
        $this->blocks = [];
        $macros["_self"] = $this->macros["_self"] = $this;
    }
```

Look at the last line of the method. PHPStan reports this error:

> Implicit array creation is not allowed - variable $macros does not exist.

It's because by default, `$macros[$name] =` is prepended in 
https://github.com/twigphp/Twig/blob/f3db1d1a0c5cc21fad255bd6690bbde16d378459/src/Node/Expression/Variable/AssignTemplateVariable.php#L31-L36

So we have 2 options:

1) add `$macros = $this->macros;` as the beginning of the constructor, just like doDisplay and block_ methods. Now PHPStan sees that `$macros` exists, and that adding something to it is fine.
2) remove the `$macros[$name] = ` prefix when inside the constructor

Reply:
I feel option 2 makes more sense, shall I try that?

Reply:
This behavior comes from `MacroAutoImportNodeVisitor`, which has been removed in 3.15, so I don't think there is anything to fix.

Reply:
Ouch... then I'll wait for 3.15 to be released. Hopefully this can be done with https://github.com/twigphp/Twig/pull/4415 included 🙏